### ISSUE TITLE
[Feature Request] Dynamic shelves?

### ISSUE THREAD

**Issue by iroQuai** [crocodilestick/Calibre-Web-Automated#44]
I don't know if this in in the scope of your project, or maybe I should open an issue in the original calibre web repo., but i'll give it a try!

I've got one calibre database with books for multiple family members. I've created a custom column called "collections" (because this way  we could add all the books to all our family e-readers and the kobo software could seperate my books from my partners books). 
With Calibre web I have no easy way of seperating those collections. 

This is what we do now: 
Me and my partner have seperate logins. We filter out eachothers books by filling in the "add allowed/denied custom column values" menu in the respective profiles (after filling in "collections" in the "View Restrictions based on Calibre column" in UI config. Works fine, but there is no access to eachothers books anymore!

I guess the public shelves would come in handy. But I can only add books manually! Most of the time forget to do this. It would be wonderful to have dynamic shelve poulation, based on search results for instance?  with advances search I can filter out only one collection).

There are probably other ways to adress this issue; some people use virtual libraries instead of collections and would run into the same problems. 

Sorry i'm using way to much word again...

**TLDR: I'm looking for an easy way to give people easy access to eachothers libraries, without cluttering their main libraries too much. Maybe with public shelves that populate automaticly (based on search parameters) or via an extra menu item based on a custom calibre column?**

**Comment by jmarmstrong1207** [comment]
I think the best way to solve this could be to have a button in the book list page where you select multiple books at once, and click a button like "add to shelf". What do you think?

**Comment by jmarmstrong1207** [comment]
Oh actually, calibre-web already has a feature to do this. Search a certain keyword, then you'll see an add to shelf button. This will add all search results to the shelf
![image](https://github.com/user-attachments/assets/947565d9-3fdb-428b-b431-af2774cfe3cd)

**Comment by iroQuai** [comment]
Searching and adding to shelf does work, but it is still a manual process every time I import one or more books. Dynamic Shelves could automate that

I just noticed an open issue on the c-w GitHub on this very topic:

https://github.com/janeczku/calibre-web/issues/2619#issue-1479851594

**Comment by jedinjapan** [comment]
Adding to this request: if you have existing shelves when coming from calibre desktop, these are not imported as shelves into calibre web. For the initial setup of calibre-web when operating as Admin, this should be at least an option.

**Comment by iroQuai** [comment]
Since this is a frequent request on the regular CW too, wouldn't this be out of scope for CWA? 

https://github.com/janeczku/calibre-web/issues/2619

**Comment by jmarmstrong1207** [comment]
> Since this is a frequent request on the regular CW too, wouldn't this be out of scope for CWA?
> 
> [janeczku/calibre-web#2619](https://github.com/janeczku/calibre-web/issues/2619)

After the main features are ironed out, I'd be open to having extra features implemented like this but it wont be worked on for a while unless another contributor helps with implementation
